### ISSUE TITLE
Table layout: index number wrong when `flip: true`

### ISSUE THREAD

**Issue by distantnative** [k-next/kirby#4386]
> The index number is not flipped when using flip: true — the pages are. It should always display the actual page index number, just like the status dialog.

Original thread: https://github.com/getkirby/kirby/issues/4374

**Comment by distantnative** [comment]
I'm a bit torn here - the table index number is actually not the `num` of pages or the `sort` of files. That would become rather problematic e.g. with page `num` based on dates etc.

On the other hand I do understand that for pages that actually use a number for `num` this could be perceived as the `num`.

**Comment by distantnative** [comment]
One option could be to actually get rid of the # column when using table in sections and put the sort handles in the preview image column on hover.

**Comment by afbora** [comment]
> One option could be to actually get rid of the # column when using table in sections and put the sort handles in the preview image column on hover.

Having counters on tables can be very useful. It would be great if it could be enabled and disabled optionally: `counter: false`

**Comment by lukasbestle** [comment]
In the list layout we never had counters. I think they only make sense if they have a semantic meaning. As Nico wrote, we cannot really use the `num`/`sort` because of the custom formats. So if we have counters, they can only be auto-increments from 1, which is really confusing. Also every new option we add makes the configuration and code more complex.

I'd vote for hiding the `#` column entirely in sections.

**Comment by afbora** [comment]
If num prop is not custom and not flip=true, how about if sort column shows up by default for table layout?

**Comment by nilshoerrmann** [comment]
A comment as a user: 

### Regarding options

Having index numbers is a great feature when dealing with tabular, spreadsheet-like data. But this section might well be used differently. So for versatility reasons it would be great to have an option in the blueprint to switch this on and off.

> Also every new option we add makes the configuration and code more complex.

From my experience, the number of options is not an issue for us a developers as long a the options are named and used consitently and are well documented. The reference on the website does a great job regarding option documentation.

> If num prop is not custom and not flip=true, how about if sort column shows up by default for table layout?

I would very much prefer many options over complex logic behind the scenes. Many options are way easier to comprehend than hidden logic that tries to guess what's best for me.

### Regarding using `num`

As long as the index numbers could be hidden, why can't you just use `num` as the column value? If the sorting is determined by a custom order id for example, the display in the table would still make sense in the context of the install.

### Regarding sorting

Pages and files sections have their drag handles outside the item itself, hovering in front of the item. Why is it bound to the table index column at all in this case?

**Comment by bastianallgeier** [comment]
To simplify this for the release next week, I would remove the index entirely for now until we come up with good options. @sylvainjule's plugin also does not offer index numbers as far as I can see. Probably for similar reasons :)

**Comment by afbora** [comment]
I guess the question now is: where to put the drag/drop handle 🙂 It seems like a pretty tough decision and maybe a little bit more complex.

**Comment by bastianallgeier** [comment]
@afbora you are right. I thought it would be more easy. I would suggest that we rethink it for the launch. Maybe we need to shift it to 3.7.1 if we don't find a proper solution.

**Comment by distantnative** [comment]
@bastianallgeier in one of my earlier versions I had this working

> put the sort handles in the preview image column on hover.

Wasn't too much work, a bit of CSS. Do you think that's an approach worth trying?

**Comment by afbora** [comment]
> put the sort handles in the preview image column on hover.

@distantnative It should always be in the first column. But you know that all columns are now dynamic. The user can disable the image column.

**Comment by bastianallgeier** [comment]
I agree that the image column isn't ideal. What if we try placing it outside of the row, like in the list?

**Comment by igregson** [comment]
Wanted to add my 2 cents here that the number always starting at 1 (even when not on the first page) is something I find super confusing and counter intuitive. If viewing results 21-40, the first item in the list should be 21 and the last should be 40.

**Comment by igregson** [comment]
Here's a screenshot to illustrate:

![CleanShot 2022-08-03 at 16 50 57@2x](https://user-images.githubusercontent.com/2054975/182626929-df2aaab4-eaf2-4d67-b19e-3805ea5b7c2a.png)

**Comment by lukasbestle** [comment]
Same suggestion to hide the number column and move the drag handle outside: https://kirby.nolt.io/460

**Comment by distantnative** [comment]
# Summary

## Sections: hide index column and move drag handle outside 
- hiding the column and moving sort handles outside isn't trivial as we are using an actual HTML table
- `overflow: visible` allows to visually move the sort handle outside the table. However, it cannot be grabbed then anymore

**Comment by nilshoerrmann** [comment]
Have you considered putting the sort handle inside its own (first) column, only making it look like it's outside the table visually?

**Comment by distantnative** [comment]
Of course, but "making it look like it's outside the table visually" is the issue: table border/shadows etc. don't work then properly anymore.

**Comment by miragecraft** [comment]
Why can’t the sort handle be grabbed when moved outside the table? Do you have a demo?

What about building the table with CSS grid and add accessibility using `aria-role`?

**Comment by distantnative** [comment]
> Why can’t the sort handle be grabbed when moved outside the table? Do you have a demo?

Tried it locally, didn't work. `overflow: visible` and then positioning left outside the table cell - as soon as I move the curator out of the table, the hover state drops and the sort handle disappears before it can be grabbed. If you find a solution that works, please let me know.

> What about building the table with CSS grid and add accessibility using aria-role?

Guess that could be an option down the road, quite a bit of work, basically rewriting the whole table component. And so far I'm too concerned about loosing accessibility support etc. I would need to read up more on that to be sure moving to a bunch of div containers would be an option.

**Comment by miragecraft** [comment]
> ... as soon as I move the curator out of the table, the hover state drops and the sort handle disappears before it can be grabbed. If you find a solution that works, please let me know.

I see what you mean now, there can't be any gaps between the table and the handle or else the hover state is dropped.

You need to build a bridge between the table and the sort handle, try using `background: transparent` to tell the browser to paint something there.

**Comment by distantnative** [comment]
Have you actually tried this?

My code:

``` css
.k-table-index-column {
	overflow: visible;
	width: 0;
	padding: 0;
	background: transparent;
}
.k-table-sort-handle {
	margin-inline-start: -1rem;
}
```

Doesn't work:

https://user-images.githubusercontent.com/3788865/191525503-a7f42dcc-b5c1-4ee5-a34d-f6a084757af6.mov

**Comment by miragecraft** [comment]
My guess is, you only added transparent background to the box with the handle in it, however the box is not touching the table, there's a gap in between. As soon as your cursor moves into that gap the hover state is dropped.

Change:

[⠿]  gap [ cell ]  

Into:

[⠿&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;][ cell ]  🠔 gap removed, padding/width added to simulate the gap, fill with transparent background

Since I don't have access to your code to try, I can only guess. If you isolate a test case and post it here (or on Codepen) I can whip up a solution for you.

This is a pretty old-school CSS problem, shouldn't be hard to solve.

**Comment by miragecraft** [comment]
Oh, I see you added `background: transparent` to the `.k-table-index-column`.

That column has zero width, there's no point adding it there because the cursor will never hover over it anyways.

You want to add it to `.k-table-sort-handle` instead and make sure there's no gap between it and the table.

Try the following, I'm using `position: absolute` and `right: 100%` to ensure it's gapless. If you need more spacing add padding to `.k-table-sort-handle`.

If you want to use logical properties, switch out `right` with `inset-inline-end` and `top` with `inset-block-start`.

```
.k-table-index-column {
	overflow: visible;
	width: 0;
	padding: 0;
	position: relative;
}

.k-table-sort-handle {
        background: transparent;
	position: absolute;
	right: 100%;
	top: 0;
}
```

Alternatively, you can use CSS transform, less code but there is no logical equivalent to `translateX` yet:

```
.k-table-index-column {
	overflow: visible;
	width: 0;
	padding: 0;
}

.k-table-sort-handle {
        background: transparent;
        transform: translateX(-100%);
}
```

**Comment by distantnative** [comment]
Thanks for trying to help! I am using the `develop` branch https://github.com/getkirby/kirby/tree/develop - only added a `k-table-items` class here to better target the specific table: https://github.com/getkirby/kirby/blob/develop/panel/src/components/Layout/Items.vue#L8


This somewhat helped: I get now as far as grabbing the handle. But as soon as I try to move, the handle disappears. Also the dragging doesn't actually work while I move outside the table (where the handle is located). Only when moving inside the table while dragging, the drag actually starts having an effect:

https://user-images.githubusercontent.com/3788865/191535843-ec933a0c-26ce-49c7-9c60-573cc211dd4c.mov

**Comment by miragecraft** [comment]
Sorry, I don't have time to do a deep dive into the code, but my guess is currently you're only showing the sort handle on hover.

However while dragging it's no longer in hover state, you need to keep the sort handle visible while in the drag state in addition to hover state.

**Comment by distantnative** [comment]
That doesn't solve the dragging issue - I think the sortable library has an issue that it's not happy when dragging in visible overflow space.

**Comment by miragecraft** [comment]
> That doesn't solve the dragging issue - I think the sortable library has an issue that it's not happy when dragging in visible overflow space.

Keep in mind that dragging needs a target, and if you're hiding the `.k-table-sort-handle` box then the area the sort handles occupy no longer counts as part of the table when dragging.

What I would try, is keep `.k-table-sort-handle` box with the transparent background visible to the browser while dragging and hide the drag handle icon using `visibility:hidden` in order to keep this space marked as part of the table.

You might consider using this method for showing/hiding the sort handle on hover as well in order to avoid code duplication.

**Comment by afbora** [comment]
@getkirby/frontend I think this is less of a bug and more of a UX/design decision about what the `#` column in the table layout is meant to represent, so it can reasonably be treated as product behavior rather than a technical defect and the issue can be closed.
